models/login_source.go

```diff
@@ -408,16 +408,12 @@ func LoginViaLDAP(user *User, login, password string, source *LoginSource, autoR
 	if err != nil {
 		return nil, err
 	} else if isExist {
-		user.FullName = composeFullName(sr.Name, sr.Surname, sr.Username)
-		user.Email = sr.Mail
-		// Change existing admin flag only if AdminFilter option is set
-		if len(source.LDAP().AdminFilter) > 0 {
-			user.IsAdmin = sr.IsAdmin
+		if user.ProhibitLogin {
+			return nil, ErrUserProhibitLogin{user.ID, user.Name}
```

Comment:
This will produce an `ErrUserProhibitLogin` encapsulated inside an `ErrUserProhibitLogin`. Perhaps you should just skip the "update" code when `user.ProhibitLogin` and let the caller decide how to handle the error.